contracts/stellar-interchain-token-service/src/abi.rs

```diff
@@ -89,13 +100,19 @@ impl Message {
                 minter: into_vec(minter).into(),
             }
             .abi_encode_params(),
-            Self::RegisterTokenMetadata(types::RegisterTokenMetadata {
-                token_address,
-                decimals,
-            }) => RegisterTokenMetadata {
-                messageType: MessageType::RegisterTokenMetadata.into(),
-                tokenAddress: token_address.to_alloc_vec().into(),
-                decimals,
+            Self::LinkToken(types::LinkToken {
+                token_id,
+                token_manager_type,
+                source_token_address,
+                destination_token_address,
+                params,
+            }) => LinkToken {
+                messageType: MessageType::LinkToken.into(),
+                tokenId: FixedBytes::<32>::new(token_id.into()),
+                tokenManagerType: token_manager_type.try_into().expect("failed to convert"),
```

Comment:
bad use of `expect` since a user provided uint256 could fail to convert to `i128`. So, we should use a more appropriate type. Unlike ITS hub which doesn't care about the token manager type, ITS edge needs to understand it, so let's use an enum for it